scripts/validate_min_versions_in_sync.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""
+Check pandas required and optional dependencies are synced across:
+
+ci/deps/actions-.*-minimum_versions.yaml
+pandas/compat/_optional.py
+
+TODO: doc/source/getting_started/install.rst
+
+This is meant to be run as a pre-commit hook - to run it manually, you can do:
+
+    pre-commit run validate-min-versions-in-sync --all-files
+"""
+from __future__ import annotations
+
+import ast
+import pathlib
+import sys
+
+DOC_PATH = pathlib.Path("doc/source/getting_started/install.rst").resolve()
+CI_PATH = next(
+    pathlib.Path("ci/deps").absolute().glob("actions-*-minimum_versions.yaml")
+)
+CODE_PATH = pathlib.Path("pandas/compat/_optional.py").resolve()
+
+
+def get_versions_from_code(content: str) -> dict[str, str]:
```

Comment:
Hm. Instead of using ast, can we just import the dict(pandas.compat._optional.VERSION)?

Reply:
When I try to add `from pandas.compat._optional import VERSIONS`, I get

```
% pre-commit run validate-min-versions-in-sync --all-files
Check minimum version of dependencies are aligned........................Failed
- hook id: validate-min-versions-in-sync
- exit code: 1

Traceback (most recent call last):
  File "scripts/validate_min_versions_in_sync.py", line 20, in <module>
    from pandas.compat._optional import VERSIONS
ModuleNotFoundError: No module named 'pandas'
Traceback (most recent call last):
  File "scripts/validate_min_versions_in_sync.py", line 20, in <module>
    from pandas.compat._optional import VERSIONS
ModuleNotFoundError: No module named 'pandas'
Traceback (most recent call last):
  File "scripts/validate_min_versions_in_sync.py", line 20, in <module>
    from pandas.compat._optional import VERSIONS
ModuleNotFoundError: No module named 'pandas'
Traceback (most recent call last):
  File "scripts/validate_min_versions_in_sync.py", line 20, in <module>
    from pandas.compat._optional import VERSIONS
ModuleNotFoundError: No module named 'pandas'
```

I think because these pre-commit hooks run in their own venv, they are not aware of the pandas directory as package

Reply:
Interesting. Maybe try to add the compat dir to ``sys.path``?
e.g.
``
>>> import sys
>>> sys.path.append("pandas/compat")
>>> import _optional
>>> _optional.VERSIONS
``

Reply:
So with this change

```
% git diff
diff --git a/scripts/validate_min_versions_in_sync.py b/scripts/validate_min_versions_in_sync.py
index 19728061e8..f6c03c099d 100755
--- a/scripts/validate_min_versions_in_sync.py
+++ b/scripts/validate_min_versions_in_sync.py
@@ -17,6 +17,11 @@ import ast
 import pathlib
 import sys

+sys.path.append("pandas/compat")
+import _optional
+
+version = _optional.VERSIONS
+
```

I get

```
% pre-commit run validate-min-versions-in-sync --all-files
Check minimum version of dependencies are aligned........................Failed
- hook id: validate-min-versions-in-sync
- exit code: 1

Traceback (most recent call last):
  File "scripts/validate_min_versions_in_sync.py", line 21, in <module>
    import _optional
  File "pandas/compat/_optional.py", line 8, in <module>
    from pandas.util.version import Version
ModuleNotFoundError: No module named 'pandas'
```

Reply:
Sorry about the error and for pushing on this issue. I've pulled your PR locally and I think I've found something that works. I'll add it as a suggestion.

Reply:
Great that worked! I agree this approach is better than parsing ASTs